semgrep/semgrep/dependency_aware_rule.py

```diff
@@ -65,12 +66,18 @@ def run_dependency_aware_rule(
     in the `targets` argument.
     """
 
-    # print(list(targets[0].parents))
+    # TODO fix this; associate targets to the lockfiles that they actually use, and remove the env var logic below
     top_level_target_rooted = list(targets[0].parents)
     top_level_target: Path = (
         targets[0] if len(top_level_target_rooted) == 0 else top_level_target_rooted[-1]
     )
-    # TODO fix this; run on the top-level of all the targets
+
+    lockfile_target_opt = getenv("SEMGREP_LOCKFILE_PATH")
+    if not lockfile_target_opt:
+        lockfile_target = top_level_target
+    else:
+        lockfile_target = Path(lockfile_target_opt)
```

Comment:
```suggestion
    lockfile_target = Path(getenv("SEMGREP_LOCKFILE_PATH", top_level_target))
```

even for just a week simpler code might help you here 🤷